### ISSUE TITLE
lagoon-core: support installing with managed mariadb, maybe even with MariaDBConsumer of dbaas 

### ISSUE THREAD

**Issue by Schnitzel** [uselagoon/lagoon-charts#86]
I'm thinking how we could install lagoon-core with the MariaDBConsumer of the https://github.com/amazeeio/dbaas-operator
as running databases in pods is not the very best system (not scalable, not HA, etc.) it would be much nicer to have a managed Database for it.

It's gonna be a bit tricky though to do this, after some research I found the lookup system of helm where we could lookup the credentials inside an `MariaDBConsumer` object, but then we need to create the `MariaDBConsumer` before all other resources. pre-install hooks of helm could do this, but they don't have really a "wait until the credentials are filled" so this might cause some issues).
Therefore I created https://github.com/amazeeio/dbaas-operator/issues/23 which would have dbaas create a secret which we then could mount in the api and keycloak services (and in my understanding k8s would not try to start the pod until the secret exists).

of course we can also in an initial phase make it all manual: just pass host, user, db, pass of the mariadb credentials in the helm file, happy to have this as a first solution

**Comment by smlx** [comment]
Yep I agree that creating the secret would be the nicest automated solution.

And yes, until that feature appears in the DBaaS operator the manual method is going to be the way to go. In fact, it is probably useful to be able to specify DB details manually anyway since there might be scenarios where you want to use another DB entirely (e.g. neither the api-db pod nor the DBaaS).
